src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs

```diff
@@ -1797,6 +1797,18 @@ static int Main()
         return (errCount > 0) ? 1 : 0;
     }
 }";
+            // the grammar does not allow a query on the right-hand-side of &&, but we allow it except in strict mode.
+            CreateCompilationWithMscorlibAndSystemCore(source, parseOptions: TestOptions.Regular.WithFeature("strict", "true")).VerifyDiagnostics(
+                // (23,26): error CS1525: Invalid expression term 'from'
+                //         var b = false && from x in src select x; // WRN CS0429
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "from x in src").WithArguments("from").WithLocation(23, 26),
```

Comment:
Of course as the language is extended in the future the parser will have to be modified as well, but if there is another valid non-query interpretation of the sequence of tokens, the current parser already gets it wrong. This change does not make that any better or worse. However since the query expression feature was carefully designed to not intersect with existing constructs, and it already performs a look-ahead to ensure it is a query syntax, it is unlikely there is a bug lurking there.